scripts/_helpers.py

```diff
@@ -1200,118 +1200,83 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
-def get_GADM_layer(country_list, layer_id, update=False, outlogging=False):
-    """
-    Function to retrieve a specific layer id of a geopackage for a selection of
-    countries.
-
-    Parameters
-    ----------
-    country_list : str
-        List of the countries
-    layer_id : int
-        Layer to consider in the format GID_{layer_id}.
-        When the requested layer_id is greater than the last available layer, then the last layer is selected.
-        When a negative value is requested, then, the last layer is requested
-    """
-    # initialization of the list of geodataframes
-    geodf_list = []
-
-    for country_code in country_list:
-        # download file gpkg
-        file_gpkg, name_file = download_GADM(country_code, update, outlogging)
-
-        # get layers of a geopackage
-        list_layers = fiona.listlayers(file_gpkg)
-
-        # get layer name
-        if layer_id < 0 | layer_id >= len(list_layers):
-            # when layer id is negative or larger than the number of layers, select the last layer
-            layer_id = len(list_layers) - 1
-        code_layer = np.mod(layer_id, len(list_layers))
-        layer_name = (
-            f"gadm36_{two_2_three_digits_country(country_code).upper()}_{code_layer}"
-        )
-
-        # read gpkg file
-        geodf_temp = gpd.read_file(file_gpkg, layer=layer_name)
-
-        # convert country name representation of the main country (GID_0 column)
-        geodf_temp["GID_0"] = [
-            three_2_two_digits_country(twoD_c) for twoD_c in geodf_temp["GID_0"]
-        ]
-
-        # create a subindex column that is useful
-        # in the GADM processing of sub-national zones
-        geodf_temp["GADM_ID"] = geodf_temp[f"GID_{code_layer}"]
+def _get_shape_col_gdf(path_to_gadm, co, gadm_level, gadm_clustering):
+    from build_shapes import get_GADM_layer
```

Comment:
Are there any specific reasons to have a local import for `get_GADM_layer`? 

That is absolutely consistent with all the other code in `_helpers`, so the comment is not quite relevant for the PR. I'm just wondering if we are not over-using local imports inside the functions. What do you think?

Reply:
Yes, otherwise there is a circular import: helpers imports build_shapes and build_shapes imports helpers and it does not work actually.
Tbh, in the future, we shall avoid the use of get_gadm_layer here potentially and use the shapes themself.
This PR however aims to revise the existing code, to avoid adding too many changes together.

Reply:
Ahh, ok, thanks for explaining! Absolutely agree that it's better to keep PR focused